presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -79,6 +80,7 @@ public Analysis analyze(Statement statement, boolean isDescribe)
         Analysis analysis = new Analysis(rewrittenStatement, parameters, isDescribe);
         StatementAnalyzer analyzer = new StatementAnalyzer(analysis, metadata, sqlParser, accessControl, session, warningCollector);
         analyzer.analyze(rewrittenStatement, Optional.empty());
+        new PostPruningTableColumnsAnalyzer(analysis).analyze(analysis.getStatement());
```

Comment:
Make the constructor private and add a static function that analyzes it - stylistically better.

Reply:
Done